rio_tiler/reader.py

```diff
@@ -256,11 +256,18 @@ def read(
         # TODO: DEPRECATED, masked array are already using bool
         if force_binary_mask:
             pass
-
+        # profile is added  only if unscaling was requested. This would ensure/signalized
+        # an advanced usage specifically applied to COGS
+        dataset_profile = None
         if unscale:
             data = data.astype("float32", casting="unsafe")
             numpy.multiply(data, dataset.scales[0], out=data, casting="unsafe")
             numpy.add(data, dataset.offsets[0], out=data, casting="unsafe")
+            dataset_profile = dataset.profile
+            if not 'scales' in dataset_profile:
+                dataset_profile['scales'] = dataset.scales
+            if not 'offsets' in dataset_profile:
+                dataset_profile['offsets'] = dataset.offsets
```

Comment:
what if we want the scales and offsets info even if we don't `apply` them?

Reply:
alos I think if we add `profile` we should always populate the field, not just when we `unscale=True`

Reply:
well, i made it conditional on unscale request as to avoid issues with STAC and Mosaic readers.
I believe the profile probably is always available but i did this per your suggestion.

It would be easy to remove the condition and update the profile with **"scales" and "offsets"**